HRIS.Services/Services/DashboardService.cs

```diff
@@ -0,0 +1,247 @@
+                TotalNumberOfEmployeesOnClients = totalNumberOfEmployeesOnClients,
+                TotalNumberOfEmployeesOnBench = totalNumberOfEmployeesOnBench.TotalNumberOfEmployeesOnBench,
+                BillableEmployeesPercentage = Math.Round(billableEmployeesPercentage, 0),
+                EmployeeTotalDifference = employeeTotalDifference,
+                isIncrease = isIncrease
+            };
+        }
+
+        public async Task<MonthlyEmployeeTotalDto> GetEmployeePreviousMonthTotal()
+        {
+            var previousMonth = DateTime.Now.AddMonths(-1).ToString("MMMM");
+
+            var previousEmployeeTotal = _db.MonthlyEmployeeTotal
+                                           .Get()
+                                           .Where(e => e.Month == previousMonth)
+                                           .FirstOrDefault();
+
+            if (previousEmployeeTotal == null) return await GetEmployeeCurrentMonthTotal();
+
+            return previousEmployeeTotal.ToDto();
+        }
+
+        public async Task<MonthlyEmployeeTotalDto> GetEmployeeCurrentMonthTotal()
+        {
+            var currentMonth = DateTime.Now.ToString("MMMM");
+
+            var currentYear = DateTime.Now.Year;
+
+            var currentEmployeeTotal = _db.MonthlyEmployeeTotal
+                                          .Get()
+                                          .Where(e => e.Month == currentMonth && e.Year == currentYear)
+                                          .FirstOrDefault();
+
+            if (currentEmployeeTotal == null)
+            {
+                var employeeTotalCount = await _db.Employee.GetAll();
+
+                var employeeCountTotalsByRole = GetEmployeeCountTotalByRole();
+
+                var monthlyEmployeeTotalDto = new MonthlyEmployeeTotalDto
+                {
+                    Id = 0,
+                    EmployeeTotal = employeeTotalCount.Count,
+                    DeveloperTotal = employeeCountTotalsByRole.DevsCount,
+                    DesignerTotal = employeeCountTotalsByRole.DesignersCount,
+                    ScrumMasterTotal = employeeCountTotalsByRole.ScrumMastersCount,
+                    BusinessSupportTotal = employeeCountTotalsByRole.BusinessSupportCount,
+                    Month = currentMonth,
+                    Year = currentYear
+                };
+                var newMonthlyEmployeeTotal = new MonthlyEmployeeTotal(monthlyEmployeeTotalDto);
+
+                return (await _db.MonthlyEmployeeTotal.Add(newMonthlyEmployeeTotal)).ToDto();
+            }
+            return currentEmployeeTotal.ToDto();
+        }
+
+        public EmployeeOnBenchDataCard GetTotalNumberOfEmployeesOnBench()
+        {
+            var totalNumberOfDevsOnBench = _db.Employee.Get()
+                                              .Where(c => c.ClientAllocated == null && c.EmployeeTypeId == 2)
+                                              .ToList().Count;
```

Comment:
```suggestion
            var totalNumberOfDevsOnBench = await _db.Employee.Get()
                                              .CountAsync(c => c.ClientAllocated == null && c.EmployeeTypeId == 2);
```

We need to use the Async call for database quieries

Do the same for below _db count calls